pkg/proxy/config.go

```diff
@@ -87,6 +93,16 @@ func NewConfig(opts *proxyoptions.Options) (*Config, error) {
 		return nil, err
 	}
 
+	c.ShardsConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: c.Options.ShardsKubeconfig},
+		// We override the Server here so that the user doesn't have to specify unused server value
+		// The Server must have HTTPS scheme otherwise CA won't be loaded (see IsConfigTransportTLS method)
```

Comment:
alternatively we could provide certificates via separate flags and construct the config manually

Reply:
That would probably be my preference, but I'm ok merging as-is and adjusting later if we decide to go that route, unless @sttts @stevekuznetsov prefer making the change now.

Reply:
It would require at least 2 flags, one for the client cert pair and the other one for the public CA's certificate.

On the other hand kubeconfig has an advantage of specifying other means of authNZ, although I'm not sure we would ever use i.e. tokens.

Reply:
+1 for merging as is.